### DISCUSSION TITLE
Trying to find the correct weights for BCs, seems not working

### DISCUSSION THREAD

**Discussion by mercury-4600** [lululxvi/deepxde#690] [General]
Hi Dr. Lu,

First, Thanks for your amazing contribution.

I would like to use DeepXDE to solve a Laplace equation on a disk, I have settled all of it's boundary conditions(it gots 3 BCs in codes) and used specific training points, it can work normally without any report errors, but still failed to get the correct solution. And so far, I have tried some methods in the issue to reduce my error, like :

- Used more specific training points and added more on BCs;
- Used larger network structure;
And I tried adding weights as you have mentioned in 
#40 
The weights I added is :

`model.compile("adam", lr=1e-3, metrics=["l2 relative error"],loss_weights=[3,1.3e-1,1.3e-1,1.3e-1,1.3e-1,7.8e-2,7.8e-2])`
The solution is still unsatisfactory, I thought its because  I didn't set a correct  distribution of weights for each item, so I added a simple traversal search method to help me find the suitable distributions:

The simple search method I added is a nested for loop：
`for w1 in range(10, 100, 10):`
     `for w2 in range(10, 100, 10):`
`         for w3 in range(10, 100, 10):`
`             model.compile("adam", lr=1e-3, metrics=["l2 relative error"], loss_weights=[w, w2,w2,w2,w2,w3,w3])`
`             losshistory, train_state = model.train(epochs= 5000)`
`             if error_max <= s:`
`                 print("Error ends at:",error_max)`
`                 print("W ends at:",w1, w2 , w2 , w2 , w2, w3 , w3 )`
The `s` is a  ideal difference from the correct solution. After few day's trying, it seems not working, because the error I got at each epoch is similar, or saying that the change steps of `w1,w2,w3` have imperceptible effect on error.  

So I wonder:
1. If this way is feasible or not, could I find a suitable weights for the BCs in order to get the correct solution?
2. What else I can do to find the correct solution?

**Comment by lululxvi** [comment]
Here is the code of Laplace equation on a disk: https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/laplace.disk.html
